swimster-api/models/Reservation.js

```diff
@@ -124,6 +122,74 @@ x
         const reservations = result.rows
         return reservations
     }
+
+    static async fetchReservationsFromUser({ user }) {
+        // fetches all reservations created by user
+        const result = await db.query(`
+            SELECT  reservations.id,
+                    reservations.date,
+                    reservations.start_time,
+                    reservations.end_time,
+                    reservations.guests,
+                    reservations.total,
+                    reservations.status,
+                    reservations.created_at,
+                    users.email,
+                    (
+                        SELECT hostUsers.email
+                        FROM users AS hostUsers
+                        WHERE hostUsers.id = (
+                            SELECT listings.host_id
+                            FROM listings
+                            WHERE listings.id = listing_id
+                        )
+                    ) AS "host_email",
+                    listings.id AS "listing_id",
+                    listings.title AS "listing_title",
+                    listings.address AS "listing_address",
+                    listings.images AS "listing_images"
+            FROM reservations
+            JOIN users ON users.id = reservations.user_id
+            JOIN listings ON listings.id = reservations.listing_id
+            WHERE user_id = (SELECT id FROM users WHERE email = $1)
```

Comment:
Do we not have access to user_id from the User object? That way we avoid the subquery that tries to get the ID from email.

`WHERE user_id = $1`